server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2156,15 +2148,26 @@ protected SegmentInfos getLatestSegmentInfos() {
      */
     @Override
     public GatedCloseable<SegmentInfos> getSegmentInfosSnapshot() {
-        final SegmentInfos segmentInfos = getLatestSegmentInfos();
+        final SegmentInfos segmentInfos;
```

Comment:
lets update the java doc for this method. The dependency order of this method and  `getLatestSegmentInfos` has reversed.